src/services/ManifoldsOverloads.jl

```diff
@@ -0,0 +1,283 @@
+
+See also: [`Jl_trivialized'](@ref), [`Jr`](@ref), `Manifolds.parallel_transport_direction`, `Manifolds.parallel_transport_to`, `Manifolds.parallel_transport_along`
+"""
+parallel_transport_best(
+  M::AbstractManifold,
+  p,
+  X::AbstractArray,
+  d::AbstractArray
+) = Manifolds.parallel_transport_direction(M, p, X, d)
+
+
+
+## ================================= Lie (a/A)djoints =================================
+## ---------------------------------- Almost generic ----------------------------------
+
+# assumes inputs are Lie algebra tangent vectors represented in matrix form
+ad(
+  M::LieGroupManifoldsPirate, 
+  A::AbstractMatrix, 
+  B::AbstractMatrix
+) = Manifolds.lie_bracket(M, A, B)
+
+
+Ad(
+  M::Union{typeof(SpecialOrthogonal(2)), typeof(SpecialOrthogonal(3))}, 
+  p, 
+  X::AbstractMatrix;
+  use_upstream::Bool = _UPSTREAM_MANIFOLDS_ADJOINT_ACTION # explict to support R&D
+) = if use_upstream
+  Manifolds.adjoint_action(M, p, X)
+else
+  p*X*(p')
+end
+
+
+function Ad(
+  M::Union{typeof(SpecialEuclidean(2)), typeof(SpecialEuclidean(3))}, 
+  p, 
+  X::ArrayPartition;
+  use_upstream::Bool = _UPSTREAM_MANIFOLDS_ADJOINT_ACTION # explict to support R&D
+)
+  if use_upstream
+    # TODO swap and test
+    Manifolds.adjoint_action(M, p, X)
+  else
+    t = p.x[1]
+    R = p.x[2]
+    v = X.x[1]
+    Ω = X.x[2]
+    # direct adjoint for SO(.)
+    RΩR = Ad(M.manifold[2], R, Ω; use_upstream )
+    ArrayPartition(-RΩR*t + R*v, RΩR)
+  end
+end
+
+
+# left trivialized Jacobian [Mahony, 2024, Theorem 4.3, matrix P_u_vee]
+# U is direction in which to transport along, given as Lie algebra element (ie tangent vector) 
+# this produces the transportMatrix P_u^vee, per [Mahony, 2024]
+# d is a Lie algebra element giving the direction of transport 
+parallel_transport_direction_lie(
+  M::typeof(SpecialOrthogonal(3)),
+  d::AbstractMatrix
+) = exp(ad(M, -0.5*d))
+
+
+# transport with 2nd order curvature approximation
+# left trivialized Jacobian [Mahony, 2024, Theorem 4.3]
+parallel_transport_along_2nd(
+  M::LieGroupManifoldsPirate,
+  p,
+  X::AbstractMatrix,
+  d::AbstractMatrix
+) = Jl_trivialized(M, d) * vee(M,p,X)
+
+
+parallel_transport_best(
+  M::LieGroupManifoldsPirate,
+  p,
+  X::AbstractMatrix,
+  d::AbstractMatrix
+) = parallel_transport_along_2nd(M,p,X,d)
+
+## ----------------------------- SpecialOrthogonal(3) -----------------------------
+
+
+# matrix versions
+# [Chirikjian, 2012 Vol2, p.39]
+ad(
+  ::typeof(SpecialOrthogonal(3)), 
+  X
+) = X
+
+Ad(
+  ::typeof(SpecialOrthogonal(3)), 
+  R
+) = R
+
+
+## For SO(3) Jr and Jl + inv closed forms, see [Chirikjian 2012, Vol2, Vol2 p.40] !!!
+
+
+# Matrix form parallel transport with curvature correction (2nd order) 
+# left trivialized Jacobian [Mahony, 2024, Theorem 4.3]
+# U is direction in which to transport (w/ 2nd order along curvature correction), 
+# Direction d is a Lie algebra element (tangent vector) providing direction of transport 
+function Jl_trivialized(
```

Comment:
Change name away from `Jl_trivialized` for more than first order Taylor expansion